src/lib/handleAuth/handleAuth.ts

```diff
@@ -35,3 +41,17 @@ export async function handleAuth({
 	}
 	throw redirect(302, url.toString());
 }
+
+const storePostRedirectUrl = (options: Record<string, string | number>) => {
+	if (options.post_redirect_url && typeof options.post_redirect_url == 'string') {
+		sessionStorage.setSessionItem(KEY_POST_REDIRECT_URL, options.post_redirect_url);
+	}
+};
+
+const redirectToPostUrl = () => {
+	if (sessionStorage.getSessionItem(KEY_POST_REDIRECT_URL)) {
+		const post_redirect_url = sessionStorage.getSessionItem(KEY_POST_REDIRECT_URL);
+		sessionStorage.removeSessionItem(KEY_POST_REDIRECT_URL);
+		throw redirect(302, kindeConfiguration.appBase + post_redirect_url);
```

Comment:
Should have caught this earlier: this has an open redirect security vulnerability. If I put in `@evil.com` as the redirect URL an attack can trick people into clicking on a link `https://customer.com/?post_redirect_url=@evil.com` and they will be redirected to `https://customer.com@evil.com` and they'll be ripe for phishing.

```suggestion
		throw redirect(302, new URL(post_redirect_url, kindeConfiguration.appBase));
```

Reply:
Thanks again for the right approach. 
Fix committed.